packages/internal/test/test-callback.js

```diff
@@ -253,3 +264,73 @@ test('isCallback', t => {
     'missing bound args',
   );
 });
+
+test('makeAttenuator', async t => {
+  const makeAttenuator = cb.prepareAttenuator(heapZone, [
+    'm0',
+    'm1',
+    'm2',
+    'm4',
+  ]);
+  const target = Far('original', {
+    m0() {
+      return 'return original.m0';
+    },
+    m1() {
+      return 'return original.m1';
+    },
+    m2() {
+      throw Error('unexpected original.m2');
+    },
+    m3() {
+      throw Error('unexpected original.m3');
+    },
+  });
+  t.throws(
+    // @ts-expect-error deliberate: m3 is not on the yeslist
+    () => makeAttenuator({ target, overrides: { m3: null } }),
+    {
+      message: `"Attenuator" overrides["m3"] not allowed by methodNames`,
+    },
+  );
+
+  // Null out a method.
+  const atE = makeAttenuator({ target, overrides: { m1: null } });
+  const p1 = atE.m0();
+  t.assert(p1 instanceof Promise);
+  t.is(await p1, 'return original.m0');
+  await t.throwsAsync(() => atE.m1(), {
+    message: `unimplemented "Attenuator" method "m1"`,
+  });
+  await t.throwsAsync(() => atE.m2(), { message: `unexpected original.m2` });
+  // @ts-expect-error deliberately attenuated out of existence
+  t.throws(() => atE.m3(), { message: /not a function/ });
+  await t.throwsAsync(() => atE.m4(), { message: /target has no method "m4"/ });
```

Comment:
Should these errors be more similar? _Can_ they be?

Reply:
The first is an engine-specific TypeError for which `not a function` is the terse message you get when `atE.m3` is not defined (in Node.js for this test).

The second is a rejection by the HandledPromise shim that contains much more specific information, such as the methods that do exist, and the type of `target`.  We could change this one, but its format is pretty deeply entrenched in Agoric SDK unit tests.